src/consumers.rs

```diff
@@ -0,0 +1,58 @@
+use std::collections::VecDeque;
+use simulators::Packet;
+
+//Consumer has a VecDeque and its buffer size
+struct Consumer {
+    vector: VecDeque<Packet>,
+    buffer_size: u32,
+}
+
+impl Consumer {
+    fn new(buffer_size: u32) -> Consumer {
+        Consumer {
+            vector: VecDeque::new(),
+            buffer_size: buffer_size,
+        }
+    }
+
+    //If the consumer doesn't have a buffer size (if it's 0), it enques
+    //IF the consumer does have a buffer size, it enques when it's length is less then it's buffer size
+    fn enqueue(&mut self, packet: Packet) -> () {
+        if self.buffer_size == 0 {
+            self.vector.push_back(packet);
+        } else if self.vector.len() < (self.buffer_size as usize) { 
```

Comment:
Saving `buffer_size` as `usize` on initialization is better than casting it each time.